pages/[id].js

```diff
@@ -50,14 +80,50 @@ export default function PlantDetails({ plants, onDeletePlant, onEditPlant }) {
     setShowEdit(false);
   }
 
-  const router = useRouter();
-  const { id } = router.query;
+  function handleAddReminder() {
+    if (newTaskType && newDueDate) {
+      onAddReminder(plant.id, newTaskType, newDueDate, repeatInterval);
+      setNewTaskType("");
+      setNewDueDate("");
+      setRepeatInterval("");
+      setShowPopup(false);
+    }
+  }
 
-  if (!router.isReady) return null;
+  function togglePopup() {
+    setShowPopup(!showPopup);
+  }
 
-  const plant = plants.find((plant) => plant.id === id);
+  function handleRepeatReminder(reminderId) {
+    const reminderEntry = reminders.find(
+      (reminder) => reminder.id === reminderId
+    );
+    if (reminderEntry && reminderEntry.interval) {
+      const newDueDate = calculateNextDueDate(
+        reminderEntry.dueDate,
+        reminderEntry.interval
+      );
+      onEditReminder(reminderId, { dueDate: newDueDate });
+    }
+  }
 
-  if (!plant) return <p>Plant not found</p>;
+  function calculateNextDueDate(currentDueDate, interval) {
+    const dueDate = new Date(currentDueDate);
+    switch (interval) {
+      case "weekly":
+        dueDate.setDate(dueDate.getDate() + 7);
+        break;
+      case "bi-weekly":
+        dueDate.setDate(dueDate.getDate() + 14);
+        break;
+      case "monthly":
+        dueDate.setMonth(dueDate.getMonth() + 1);
+        break;
+      default:
+        break;
+    }
+    return dueDate.toISOString().split("T")[0];
+  }
```

Comment:
As this is a helper (utility) function, you could move this outside the component into a separate file (in a `utils` folder for example).

Reply:
thanks, for now we don't need it. when we have more helper functions, we will do that